moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -55,11 +55,12 @@ planning_pipeline::PlanningPipeline::PlanningPipeline(const robot_model::RobotMo
   : nh_(nh), robot_model_(model)
 {
   std::string planner;
-  if (nh_.getParam(planner_plugin_param_name, planner))
+  ros::NodeHandle cnh = planning_interface::getConfigNodeHandle(nh);
```

Comment:
I still don't really follow why `getConfigNodeHandle()` is being used here. If a default pipeline is configured, `getConfigNodeHandle` will always return a node handle in the namespace of the default pipeline. With this change, non-default pipelines wouldn't be loaded anymore. If anything, the "fixed" namespace should be passed with the node handle from the constructor.

Reply:
`default_planning_pipeline` is a parameter introduced by your new config scheme via #2127. If this *and* the corresponding pipeline namespace `move_group/planning_pipelines/$default_pipeline` exist, `getConfigNodeHandle()` will return a corresponding NodeHandle, thus using the config from the new scheme. Otherwise, it just falls back to the originally passed (private) move_group namespace, thus resorting to the old/existing scheme.

> With this change, non-default pipelines wouldn't be loaded anymore.

The parameter `default_planning_pipeline` was newly introduced in #2127 and **Melodic** doesn't know about this parameter (yet). So what exactly do you mean by loading non-default pipelines?
So far, Melodic can only load a single pipeline, the one configured in the private move_group namespace.
This PR allows using the new config scheme with multiple namespaces below `move_group/planning_pipelines/*`, but only a single one can be used at a time. I explicitly didn't want to support switching.

> The "fixed" namespace should be passed with the node handle from the constructor

What do you mean by _fixed_ namespace?

Reply:
> I explicitly didn't want to support switching.

Oh, I didn't get that. So you are intentionally only allowing a single default pipeline instance. I'm fine with that if we really don't need that feature in melodic.

> What do you mean by fixed namespace?

With the assumption that melodic should support multiple pipelines, I thought that the namespace would better be handled by the code calling the pipeline constructor. With "fixed" I meant the switching of the namespace inside `getConfigNodeHandle()`.

Reply:
> Oh, I didn't get that.

I was afraid that you misconceived my intention... Good, that we are on the same page now.
Do you see other issues or do you approve?

Reply:
Looks good to me, thanks!